lib/tasks/verify_application_api.rake

```diff
@@ -0,0 +1,104 @@
+class MeasureSerializingApplications
```

Comment:
Im in two minds around committing this code to the repo. As this is mostly a list of profiling tasks which requires a gem to be added that isn't included in this PR.

I have a few questions:
1) Do we find that we will be using these tasks quite often or will this be a one off?
2) Do we have precedence of committing benchmark tooling to the codebase? (eg `'rack-mini-profiler'`)

Happy to hear your thoughts @tomas-stefano @steventux @carlosmartinez

Reply:
That's a good point! I think this could be a one-off. I am happy to remove the rake task if everybody agrees.

By the way, I don't know the process of improving performance & scalability but the way that I discover the problem was with the ruby-prof when I saw it was doing million of calls to Psych trying to handle YAML files and doing millions of Hash transformations. But ruby-prof can be seen as "too much" to add to an app, that's the reason I didn't add. The  Bullet gem helped to see that it was doing a bunch of unnecessary queries to the DB. The bullet gem is included in the Gemfile and I wonder if there is any place people are checking the N+1 queries in the app?

Happy to hear the thoughts of everyone and remove the rake task. ✌️

Reply:
I don't mind the presence of performance related rake tasks, perhaps they could live in a `performance` namespace instead of or wrapped around `api`.

We do include `rack-mini-profiler` in our bundle, would including `ruby-prof` in `development` and `test` environments be a bad thing?

Reply:
I think (and having chatted to @carlosmartinez in person) best option as @steventux suggested is to move it out of this PR. However I think we can leave it out for now, and if we find we keep revisiting improving the performance and its not just a one off for this endpoint, we can revisit this code/pr and bring back the rake tasks and add the gem with the relevant documentation. How does that sound?